homeassistant/components/pi_hole/config_flow.py

```diff
@@ -50,47 +50,31 @@ async def async_step_init(self, user_input, is_import=False):
         errors = {}
 
         if user_input is not None:
+            host = (
+                user_input.get(CONF_HOST)
+                if is_import
+                else f"{user_input.get(CONF_HOST)}:{user_input.get(CONF_PORT)}"
+            )
+            name = user_input.get(CONF_NAME)
```

Comment:
```suggestion
            name = user_input[CONF_NAME]
```

For everything that is `vol.Required` please use the above as the key should never be missing